### DISCUSSION TITLE
`classes` argument to some widgets (like `VContainer`)

### DISCUSSION THREAD

**Discussion by kerim371** [Kitware/trame#577] [General] [answered]
Hi,

In Trame tutorials I can see that `VContainer` is used like:
```python
with SinglePageLayout(server) as layout:
    layout.title.set_text("Hello trame")

    with layout.content:
        with vuetify.VContainer(
            fluid=True,
            classes="pa-0 fill-height",
        ):
            view = vtk.VtkLocalView(renderWindow)
```

I feel like keyword `classes` is used for the styling purpose but I cant understand what `pa-0` and `fill-height` mean?
More important where can I see what other values (besides `pa-0` and `fill-heigh`) are available to define my own `classes`?
For example `VExpansionPanel` without defined `classes` is not visible at all.

**Comment by patrickoleary** [comment, marked as answer]
Go to https://vuetifyjs.com/en/styles/ it's a 404, 

<img width="1295" alt="Screenshot 2024-08-18 at 6 13 02 AM" src="https://github.com/user-attachments/assets/d088fa70-79f6-4bdc-8447-fbadadc079e0">

In the side container,  'Click' on Styles and animations to see the API and the Utility classes

<img width="291" alt="Screenshot 2024-08-18 at 6 15 16 AM" src="https://github.com/user-attachments/assets/20e3af13-0de1-4d56-8680-08b34de7d418">

For example, if you click on Sizing you will find the classes and explanations of these classes.

<img width="1285" alt="Screenshot 2024-08-18 at 6 17 04 AM" src="https://github.com/user-attachments/assets/2fe06f6d-81e9-41f5-a388-36ad3cc3c930">

for pa and more - https://vuetifyjs.com/en/styles/spacing/#how-it-works
for fill-height and more - https://vuetifyjs.com/en/styles/sizing/#usage

I have this Vuetify page always open while I'm developing. It is the GOTO place for everything Vuetify.

Finally, trame has some basic rules for working with Vuetify. That information is available here

https://kitware.github.io/trame/guide/tutorial/html.html

I have this page bookmarked for quick access.

**Reply by kerim371** [reply]
Hi,

Thank you very much for the information!
That fully describes what I was looking for!
